surefire-providers/surefire-junit-platform/src/test/java/org/apache/maven/surefire/junitplatform/JUnitPlatformProviderTest.java

```diff
@@ -865,6 +866,7 @@ void testAlwaysSkipped()
             throw new IllegalStateException( "this test should be never called" );
         }
 
+        @org.junit.jupiter.api.DisplayName( "Fails twice" )
```

Comment:
well, okay it cover the test scenario but since we made two changes (array indexes and added HashSet) we should assert the display names in the listener too.

What if we coment the DisplayName? It will not rerun the tests as we found it in Jira?

Reply:
> What if we coment the DisplayName? It will not rerun the tests as we found it in Jira?

I'm not sure I follow. I've updated the test to print the failing test names if that's what you're looking for.

Reply:
I meant something like this https://github.com/apache/maven-surefire/pull/249/commits/4f16fcde178c95d170e3d883bbf6fc0ab28998fb#diff-183e6d2cde226a9744aa928c54694b2fR239 in the tests for this bug.

Reply:
I guess you added these assertion statements for handled disnpaly names. :-)